File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/backend/app/contracts/studySessionContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field

StudySessionType = Literal["STUDY","REVIEW","PRACTICE","MOCK_EXAM"]

class StudySessionStartContract(BaseModel):
    contractName: Literal["StudySessionStart.v1"] = "StudySessionStart.v1"
    sessionType: StudySessionType = "STUDY"
    notes: str | None = Field(default=None, max_length=2000)

class StudySessionItemContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    contractName: Literal["StudySessionItem.v1"] = "StudySessionItem.v1"
    studySessionItemId: UUID
    studySessionId: UUID
    studyScopeItemId: UUID
    status: Literal["PENDING","IN_PROGRESS","COMPLETED","SKIPPED"]
    timeSpentSeconds: int
    startedAt: datetime | None
    completedAt: datetime | None
    notes: str | None
    createdAt: datetime
    updatedAt: datetime

class StudySessionContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    contractName: Literal["StudySession.v1"] = "StudySession.v1"
    studySessionId: UUID
    studyScopeId: UUID
    studentId: UUID
    sessionType: StudySessionType
    status: Literal["IN_PROGRESS","COMPLETED","CANCELLED"]
    startedAt: datetime
    endedAt: datetime | None
    notes: str | None
    createdAt: datetime
    updatedAt: datetime

class StudySessionViewContract(BaseModel):
    contractName: Literal["StudySessionView.v1"] = "StudySessionView.v1"
    session: StudySessionContract
    items: list[StudySessionItemContract]

File: apps/studentWeb/src/contracts/studySessionContract.ts
export type StudySessionType = 'STUDY' | 'REVIEW' | 'PRACTICE' | 'MOCK_EXAM'

export interface StudySessionContract {
  contractName: 'StudySession.v1'
  studySessionId: string
  studyScopeId: string
  studentId: string
  sessionType: StudySessionType
  status: 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED'
  startedAt: string
  endedAt: string | null
  notes: string | null
  createdAt: string
  updatedAt: string
}

export interface StudySessionItemContract {
  contractName: 'StudySessionItem.v1'
  studySessionItemId: string
  studySessionId: string
  studyScopeItemId: string
  status: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED' | 'SKIPPED'
  timeSpentSeconds: number
  startedAt: string | null
  completedAt: string | null
  notes: string | null
  createdAt: string
  updatedAt: string
}

export interface StudySessionViewContract {
  contractName: 'StudySessionView.v1'
  session: StudySessionContract
  items: StudySessionItemContract[]
}
